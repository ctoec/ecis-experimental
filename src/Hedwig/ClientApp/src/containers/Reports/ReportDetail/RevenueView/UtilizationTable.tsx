import React from 'react';
import { Table, TableProps, useGenericContext, FormContext } from '@ctoec/component-library';
import {
	CdcReport,
} from '../../../../generated';
import {
	prettyAge,
	prettyFundingTime,
	getReportingPeriodWeeks,
} from '../../../../utils/models';
import currencyFormatter from '../../../../utils/currencyFormatter';
import cx from 'classnames';
import {
	makePrefixerFunc,
	ReimbursementRateLine,
} from '../../../../utils/utilizationTable';
import { RevenueDataRow, getRevenueDataRows } from '../utils/getRevenueDataRows';

type UtilizationTableRow = RevenueDataRow & {
	showFundingTime?: boolean;
	// Used for numeral alignment formatting
	maxes: {
		total: number;
		rate: number;
		balance: number;
	};
}

export default function UtilizationTable() {
	const { data: report } = useGenericContext<CdcReport>(FormContext);

	// Transform RevenueDataRow into UtilizationTableRow
	let rows: UtilizationTableRow[] = getRevenueDataRows(report).map((row) => ({
		...row,
		showFundingTime: (report.organization?.fundingSpaces || []).filter((fs) => fs.ageGroup === row.ageGroup).length > 1,
		maxes: {
			total: row.total,
			balance: row.balance,
			rate: Math.max(row.ptRate || 0, row.ftRate || 0)
		}
	}));

	// And do reduce transformation for total row
	let totalRow = rows[rows.length - 1];
	totalRow = rows.reduce(
		(total, row) => ({
			...total,
			maxes: {
				total: Math.max(total.total, row.total),
				rate: Math.max(total.maxes.rate, row.ftRate || 0, row.ptRate || 0),
				balance: Math.max(Math.abs(total.balance), Math.abs(row.balance))
			}
		}),
		totalRow
	)

	// Get prefixers
	const reimbursementPrefixer = makePrefixerFunc(totalRow.maxes.rate);
	const totalPrefixer = makePrefixerFunc(totalRow.maxes.total);
	const balancePrefixer = makePrefixerFunc(totalRow.maxes.balance);

	// Instantiate table props
	const tableProps: TableProps<UtilizationTableRow> = {
		id: 'utilization-table',
		data: rows,
		rowKey: (row) => row.key,
		columns: [
			{
				name: '',
				cell: ({ row }) => (
					<th>
						<strong>{row.key === 'total' ? 'Total' : `${prettyAge(row.ageGroup)}`}</strong>
						{row.fundingTime && row.showFundingTime && (
							<> &ndash; {prettyFundingTime(row.fundingTime, { splitTimeText: 'pt/ft split' })}</>
						)}
					</th>
				),
			},
			{
				name: 'Utilization',
				cell: ({ row }) => (
					<td
						className={
							(row.key === 'total' ? 'oec-table__cell--strong' : '') +
							' ' +
							(row.count !== row.capacity ? 'oec-table__cell--red' : '')
						}
					>
						{row.count}/{row.capacity} spaces
					</td>
				),
			},
			{
				name: 'Reimbursement rate',
				className: 'text-right',
				cell: ({ row }) => {
					const { partWeeks, fullWeeks } = row.weeksSplit || {};
					return (
						<td className="text-tabular text-right">
							<ReimbursementRateLine
								prefix={reimbursementPrefixer(row.ptRate || 0)}
								prettyRate={currencyFormatter(row.ptRate || 0, true)}
								weeksInPeriod={partWeeks}
								suffix="(pt)"
							/>
							<ReimbursementRateLine
								prefix={reimbursementPrefixer(row.ftRate || 0)}
								prettyRate={currencyFormatter(row.ftRate || 0, true)}
								weeksInPeriod={fullWeeks}
								suffix="(ft)"
							/>
						</td>
					);
				},
			},
			{
				name: `Total (${getReportingPeriodWeeks(report.reportingPeriod)} weeks)`,
				className: 'text-right',
				cell: ({ row }) => {
					const prefix = totalPrefixer(row.total);
					return (
						<td
							className={cx(
								{ 'oec-table__cell--strong': row.key === 'total' },
								'text-tabular',
								'text-right'
							)}
						>
							<span>$ </span>
							<span style={{ visibility: 'hidden' }}>{prefix}</span>
							{currencyFormatter(row.total, true)}
						</td>
					);
				},
			},
			{
				name: 'Balance',
				className: 'text-right',
				cell: ({ row }) => {
					const prefix = balancePrefixer(row.balance);
					return (
						<td
							className={cx(
								{ 'oec-table__cell--strong': row.key === 'total' },
								{ 'oec-table__cell--red': row.balance < 0 },
								'text-tabular',
								'text-right'
							)}
						>
							<div className={cx('position-relative', { 'one-half-char-right': row.balance < 0 })}>
								{row.balance < 0 ? '(' : ''}
								<span>$ </span>
								<span style={{ visibility: 'hidden' }}>{prefix}</span>
								{currencyFormatter(Math.abs(row.balance), true)}
								{row.balance < 0 ? ')' : ''}
							</div>
						</td>
					);
				},
			},
		],
	};

	return (
		<>
			<h2 className="margin-bottom-neg-205">Monthly utilization</h2>
			<Table {...tableProps} fullWidth />
		</>
	);
}
