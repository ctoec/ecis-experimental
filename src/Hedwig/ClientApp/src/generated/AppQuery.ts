/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: AppQuery
// ====================================================

export interface AppQuery_me_reports {
  __typename: "CdcReportType";
  id: number;
  submittedAt: OECDate | null;
}

export interface AppQuery_me {
  __typename: "UserType";
  firstName: string;
  reports: AppQuery_me_reports[];
}

export interface AppQuery {
  me: AppQuery_me | null;
}
