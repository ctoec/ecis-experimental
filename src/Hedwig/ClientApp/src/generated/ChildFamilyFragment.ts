/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: ChildFamilyFragment
// ====================================================

export interface ChildFamilyFragment_family_determinations {
	__typename: 'FamilyDeterminationType';
	id: number;
	numberOfPeople: number;
	income: OECDecimal;
	determined: OECDate;
}

export interface ChildFamilyFragment_family {
	__typename: 'FamilyType';
	id: number;
	addressLine1: string | null;
	addressLine2: string | null;
	town: string | null;
	state: string | null;
	zip: string | null;
	homelessness: boolean;
	determinations: ChildFamilyFragment_family_determinations[];
}

export interface ChildFamilyFragment {
	__typename: 'ChildType';
	id: string;
	family: ChildFamilyFragment_family | null;
}
