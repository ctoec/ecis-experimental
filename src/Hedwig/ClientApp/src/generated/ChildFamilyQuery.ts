/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildFamilyQuery
// ====================================================

export interface ChildFamilyQuery_child_family_determinations {
	__typename: 'FamilyDeterminationType';
	id: number;
	numberOfPeople: number;
	income: OECDecimal;
	determined: OECDate;
}

export interface ChildFamilyQuery_child_family {
	__typename: 'FamilyType';
	id: number;
	addressLine1: string | null;
	addressLine2: string | null;
	town: string | null;
	state: string | null;
	zip: string | null;
	homelessness: boolean;
	determinations: ChildFamilyQuery_child_family_determinations[];
}

export interface ChildFamilyQuery_child {
	__typename: 'ChildType';
	id: string;
	family: ChildFamilyQuery_child_family | null;
}

export interface ChildFamilyQuery {
	child: ChildFamilyQuery_child | null;
}

export interface ChildFamilyQueryVariables {
	id: string;
}
