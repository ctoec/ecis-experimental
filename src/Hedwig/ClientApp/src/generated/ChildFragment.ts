/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { Gender, Age, FundingSource, FundingTime } from './globalTypes';

// ====================================================
// GraphQL fragment: ChildFragment
// ====================================================

export interface ChildFragment_enrollments_site {
	__typename: 'SiteType';
	id: number;
	name: string;
}

export interface ChildFragment_enrollments_fundings {
	__typename: 'FundingType';
	id: number;
	source: FundingSource;
	time: FundingTime;
}

export interface ChildFragment_enrollments {
	__typename: 'EnrollmentType';
	id: number;
	entry: OECDate | null;
	exit: OECDate | null;
	age: Age | null;
	site: ChildFragment_enrollments_site;
	fundings: ChildFragment_enrollments_fundings[];
}

export interface ChildFragment_family_determinations {
	__typename: 'FamilyDeterminationType';
	id: number;
	numberOfPeople: number;
	income: OECDecimal;
	determined: OECDate;
}

export interface ChildFragment_family {
	__typename: 'FamilyType';
	id: number;
	addressLine1: string | null;
	addressLine2: string | null;
	town: string | null;
	state: string | null;
	zip: string | null;
	homelessness: boolean;
	determinations: ChildFragment_family_determinations[];
}

export interface ChildFragment {
	__typename: 'ChildType';
	id: string;
	sasid: string | null;
	firstName: string;
	middleName: string | null;
	lastName: string;
	suffix: string | null;
	birthdate: OECDate | null;
	birthTown: string | null;
	birthState: string | null;
	birthCertificateId: string | null;
	americanIndianOrAlaskaNative: boolean;
	asian: boolean;
	blackOrAfricanAmerican: boolean;
	nativeHawaiianOrPacificIslander: boolean;
	white: boolean;
	hispanicOrLatinxEthnicity: boolean;
	gender: Gender | null;
	foster: boolean;
	enrollments: ChildFragment_enrollments[];
	family: ChildFragment_family | null;
}
