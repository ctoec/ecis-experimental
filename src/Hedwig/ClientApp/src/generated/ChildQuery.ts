/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildQuery
// ====================================================

export interface ChildQuery_child_family {
	__typename: 'FamilyType';
	id: number;
	caseNumber: number | null;
}

export interface ChildQuery_child {
	__typename: 'ChildType';
	id: string;
	firstName: string;
	middleName: string | null;
	lastName: string;
	suffix: string | null;
	family: ChildQuery_child_family | null;
}

export interface ChildQuery {
	child: ChildQuery_child | null;
}

export interface ChildQueryVariables {
	id: string;
}
