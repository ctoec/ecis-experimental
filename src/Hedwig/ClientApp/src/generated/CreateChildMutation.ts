/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { Gender, Age, FundingSource, FundingTime } from './globalTypes';

// ====================================================
// GraphQL mutation operation: CreateChildMutation
// ====================================================

export interface CreateChildMutation_createChildWithSiteEnrollment_enrollments_site {
	__typename: 'SiteType';
	id: number;
	name: string;
}

export interface CreateChildMutation_createChildWithSiteEnrollment_enrollments_fundings {
	__typename: 'FundingType';
	id: number;
	source: FundingSource;
	time: FundingTime;
}

export interface CreateChildMutation_createChildWithSiteEnrollment_enrollments {
	__typename: 'EnrollmentType';
	id: number;
	entry: OECDate | null;
	exit: OECDate | null;
	age: Age | null;
	site: CreateChildMutation_createChildWithSiteEnrollment_enrollments_site;
	fundings: CreateChildMutation_createChildWithSiteEnrollment_enrollments_fundings[];
}

export interface CreateChildMutation_createChildWithSiteEnrollment_family_determinations {
	__typename: 'FamilyDeterminationType';
	id: number;
	numberOfPeople: number;
	income: OECDecimal;
	determined: OECDate;
}

export interface CreateChildMutation_createChildWithSiteEnrollment_family {
	__typename: 'FamilyType';
	id: number;
	addressLine1: string | null;
	addressLine2: string | null;
	town: string | null;
	state: string | null;
	zip: string | null;
	homelessness: boolean;
	determinations: CreateChildMutation_createChildWithSiteEnrollment_family_determinations[];
}

export interface CreateChildMutation_createChildWithSiteEnrollment {
	__typename: 'ChildType';
	id: string;
	sasid: string | null;
	firstName: string;
	middleName: string | null;
	lastName: string;
	suffix: string | null;
	birthdate: OECDate | null;
	birthTown: string | null;
	birthState: string | null;
	birthCertificateId: string | null;
	americanIndianOrAlaskaNative: boolean;
	asian: boolean;
	blackOrAfricanAmerican: boolean;
	nativeHawaiianOrPacificIslander: boolean;
	white: boolean;
	hispanicOrLatinxEthnicity: boolean;
	gender: Gender | null;
	foster: boolean;
	enrollments: CreateChildMutation_createChildWithSiteEnrollment_enrollments[];
	family: CreateChildMutation_createChildWithSiteEnrollment_family | null;
}

export interface CreateChildMutation {
	createChildWithSiteEnrollment: CreateChildMutation_createChildWithSiteEnrollment | null;
}

export interface CreateChildMutationVariables {
	sasid?: string | null;
	firstName: string;
	middleName?: string | null;
	lastName: string;
	suffix?: string | null;
	birthdate?: string | null;
	birthCertificateId?: string | null;
	birthTown?: string | null;
	birthState?: string | null;
	gender?: Gender | null;
	americanIndianOrAlaskaNative?: boolean | null;
	asian?: boolean | null;
	blackOrAfricanAmerican?: boolean | null;
	nativeHawaiianOrPacificIslander?: boolean | null;
	white?: boolean | null;
	hispanicOrLatinxEthnicity?: boolean | null;
	foster?: boolean | null;
	siteId: number;
}
