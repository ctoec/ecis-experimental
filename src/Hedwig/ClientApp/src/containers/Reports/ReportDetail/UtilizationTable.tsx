import React from 'react';
import Table, { TableProps } from '../../../components/Table/Table';
import { CdcReport, Age, FundingTime, Region } from '../../../generated';
import idx from 'idx';
import moment from 'moment';
import { CdcRates } from './CdcRates';
import { prettyAge } from '../../../utils/ageGroupUtils';
import { prettyFundingTime } from '../../../utils/fundingTimeUtils';
import currencyFormatter from '../../../utils/currencyFormatter';

export function calculateRate(accredited: boolean, titleI: boolean, region: Region, ageGroup: Age, time: FundingTime) {
  const rate = CdcRates.find(rate =>
    rate.accredited === accredited &&
    rate.titleI === titleI &&
    rate.region === region &&
    rate.ageGroup === ageGroup &&
    rate.time === time
  );

  return rate ? rate.rate : 0;
}

interface UtilizationTableRow {
  key: string
  ageGroup?: Age
  fundingTime?: FundingTime
  count: number
  capacity: number
  rate?: number
  total: number
  balance: number
}

export default function UtilizationTable(report: CdcReport) {
  const periodStart = idx(report, _ => _.reportingPeriod.periodStart);
  const periodEnd = idx(report, _ => _.reportingPeriod.periodEnd);
  const weeksInPeriod = periodStart && periodEnd ? moment(periodEnd).add(1, 'day').diff(periodStart, 'weeks') : 0;

  const accredited = report.accredited;
  const fundingSpaces = idx(report, _ => _.organization.fundingSpaces) || [];

  const sites = idx(report, _ => _.organization.sites) || [];
  const ageGroups = [Age.InfantToddler, Age.Preschool, Age.SchoolAge];
  const fundingTimes = [FundingTime.Full, FundingTime.Part];

  const rows: UtilizationTableRow[] = sites.flatMap(site => {
    const region = site.region;
    const titleI = site.titleI;

    const enrollments = site.enrollments.flatMap(enrollment => {
      const cdcFunding = enrollment.fundings.find(funding => funding.source === report.type);
      if (!cdcFunding) { return []; }
      return {
        ageGroup: enrollment.ageGroup,
        time: cdcFunding.time,
      };
    });

    return ageGroups.flatMap(ageGroup => fundingTimes.flatMap(fundingTime => {
      const count = enrollments
        .filter(enrollment => enrollment.ageGroup === ageGroup && enrollment.time === fundingTime)
        .length;

      const capacity = idx(
        fundingSpaces.find(space => space.ageGroup === ageGroup && space.time === fundingTime),
        _ => _.capacity
      ) || 0;

      const cappedCount = Math.min(count, capacity);

      const rate = calculateRate(accredited, titleI, region, ageGroup, fundingTime);

      const total = cappedCount * rate * weeksInPeriod;
      const paid = capacity * rate * weeksInPeriod;
      const balance = total - paid;

      return {
        key: `${ageGroup}-${fundingTime}`,
        ageGroup,
        fundingTime,
        count,
        capacity,
        rate,
        total,
        balance,
      }
    }));
  });

  const totalRow = rows.reduce((total, row) => {
    return {
      ...total,
      count: total.count + row.count,
      capacity: total.capacity + row.capacity,
      total: total.total + row.total,
      balance: total.balance + row.balance,
    };
  }, { key: 'total', count: 0, capacity: 0, total: 0, balance: 0 });

  rows.push(totalRow);

  const tableProps: TableProps<UtilizationTableRow> = {
    id: 'utilization-table',
    data: rows.filter(row => row.count || row.capacity),
    rowKey: row => row.key,
    columns: [
      {
        name: '',
        cell: ({ row }) => (
          <th>
            <strong>
              {row.key === 'total' ? 'Total' : `${prettyAge(row.ageGroup)}`}
            </strong>
            {row.fundingTime && (
              <> &ndash; {prettyFundingTime(row.fundingTime)}</>
            )}
          </th>
        )
      },
      {
        name: 'Utilization',
        cell: ({ row }) => (
          <td className={
            (row.key === "total" ? "oec-table__cell--strong" : "") + " " +
            (row.count !== row.capacity ? "oec-table__cell--red" : "")
          }>
            {row.count}/{row.capacity} spaces
          </td>
        )
      },
      {
        name: <>Reimbursement <br />rate (weekly)</>,
        cell: ({ row }) => (
          <td>
            {row.key !== 'total' && currencyFormatter(row.rate || 0)}
          </td>
        )
      },
      {
        name: `Total (${weeksInPeriod} weeks)`,
        cell: ({ row }) => (
          <td className={row.key === "total" ? "oec-table__cell--strong" : ""}>
            {currencyFormatter(row.total)}
          </td>
        )
      },
      {
        name: `Balance`,
        cell: ({ row }) => (
          <td className={
            (row.key === "total" ? "oec-table__cell--strong" : "") + " " +
            (row.balance < 0 ? "oec-table__cell--red" : "")
          }>
            {row.balance < 0 ? '(' : ''}
            {currencyFormatter(Math.abs(row.balance))}
            {row.balance < 0 ? ')' : ''}
          </td>
        )
      }
    ]
  }

  return <>
    <h2 className="margin-bottom-neg-205">Monthly utilization</h2>
    <Table {...tableProps} fullWidth />
  </>
}
