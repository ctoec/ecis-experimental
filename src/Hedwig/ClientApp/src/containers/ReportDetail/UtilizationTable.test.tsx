import React from 'react';
import { mount } from 'enzyme';
import { Age, FundingTime, CdcReport, FundingSource, Enrollment, Region } from '../../generated';
import UtilizationTable, { calculateRate } from './UtilizationTable';
import emptyGuid from '../../utils/emptyGuid';

describe('calculateRate', () => {
  it('includes all possible rates', () => {
    [true, false].forEach(accredited => {
      [true, false].forEach(titleI => {
        [Region.East, Region.NorthCentral, Region.NorthWest, Region.SouthCentral, Region.SouthWest].forEach(region => {
          [Age.InfantToddler, Age.Preschool, Age.SchoolAge].forEach(ageGroup => {
            [FundingTime.Full, FundingTime.Part].forEach(time => {
              const rate = calculateRate(accredited, titleI, region, ageGroup, time);
              expect(rate).toBeGreaterThan(0);
            });
          });
        });
      });
    });
  });
});

const reportWithEnrollments = (enrollments: Enrollment[]) => {
  const report: CdcReport = {
    id: 1,
    organizationId: 1,
    accredited: true,
    type: FundingSource.CDC,
    reportingPeriod: {
      id: 1,
      type: FundingSource.CDC,
      period: new Date("2019-09-01"),
      dueAt: new Date("2019-10-15"),
      periodStart: new Date("2019-09-01"),
      periodEnd: new Date("2019-09-28"),
    },
    organization: {
      id: 1,
      name: "Test Organization",
      fundingSpaces: [
        {
          source: FundingSource.CDC,
          ageGroup: Age.Preschool,
          time: FundingTime.Full,
          capacity: 2,
          organizationId: 1
        },
      ],
      sites: [{
        name: "Test Site",
        region: Region.East,
        titleI: false,
        enrollments,
        organizationId: 1,
      }],
    },
  };

  return report;
};

const defaultReport = reportWithEnrollments([
  {
    id: 1,
    ageGroup: Age.Preschool,
    siteId: 1,
    childId: emptyGuid(),
    fundings: [{
      id: 1,
      source: FundingSource.CDC,
      time: FundingTime.Full,
      enrollmentId: 1,
    }],
  }
]);

describe('UtilizationTable', () => {
  it('matches snapshot', () => {
    const table = mount(<UtilizationTable {...defaultReport} />);
    expect(table.html()).toMatchSnapshot();
  });

  it('includes a row for each type of enrollment and funding space', () => {
    const report = reportWithEnrollments([
      {
        id: 1,
        ageGroup: Age.InfantToddler,
        siteId: 1,
        childId: emptyGuid(),
        fundings: [{
          id: 1,
          source: FundingSource.CDC,
          time: FundingTime.Full,
          enrollmentId: 1
        }],
      },
      {
        id: 2,
        ageGroup: Age.InfantToddler,
        siteId: 1,
        childId: emptyGuid(),
        fundings: [{
          id: 1,
          source: FundingSource.CDC,
          time: FundingTime.Part,
          enrollmentId: 1
        }],
      },
    ]);

    const table = mount(<UtilizationTable {...report} />);

    expect(table.find('tr')).toHaveLength(5);
  });

  it('does not include enrollments without an age', () => {
    const report = reportWithEnrollments([
      {
        id: 1,
        ageGroup: undefined,
        siteId:1,
        childId: emptyGuid(),
        fundings: [{
          id: 1,
          source: FundingSource.CDC,
          time: FundingTime.Full,
          enrollmentId: 1
        }],
      }
    ]);

    const table = mount(<UtilizationTable {...report} />);

    expect(table.find('td').first().text()).toEqual("0/2 spaces")
  });
});
