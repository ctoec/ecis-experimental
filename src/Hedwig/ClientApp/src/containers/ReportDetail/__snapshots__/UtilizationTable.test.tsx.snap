// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilizationTable matches snapshot 1`] = `"<h2 class=\\"margin-bottom-neg-205\\">Monthly utilization</h2><table id=\\"utilization-table\\" class=\\"oec-table oec-table--full-width\\"><thead><tr><th scope=\\"col\\" class=\\"oec-table__column-header\\" role=\\"columnheader\\" aria-sort=\\"none\\"><span class=\\"oec-table__column-title\\"></span></th><th scope=\\"col\\" class=\\"oec-table__column-header\\" role=\\"columnheader\\" aria-sort=\\"none\\"><span class=\\"oec-table__column-title\\">Utilization</span></th><th scope=\\"col\\" class=\\"oec-table__column-header\\" role=\\"columnheader\\" aria-sort=\\"none\\"><span class=\\"oec-table__column-title\\">Weekly rate</span></th><th scope=\\"col\\" class=\\"oec-table__column-header\\" role=\\"columnheader\\" aria-sort=\\"none\\"><span class=\\"oec-table__column-title\\">Total (4 weeks)</span></th><th scope=\\"col\\" class=\\"oec-table__column-header\\" role=\\"columnheader\\" aria-sort=\\"none\\"><span class=\\"oec-table__column-title\\">Balance</span></th></tr></thead><tbody><tr><th><strong>Preschool</strong> â€“ full time</th><td class=\\" oec-table__cell--red\\">1/2 spaces</td><td>$165.32</td><td class=\\"\\">$661.28</td><td class=\\" oec-table__cell--red\\">($661.28)</td></tr><tr><th><strong>Total</strong></th><td class=\\"oec-table__cell--strong oec-table__cell--red\\">1/2 spaces</td><td></td><td class=\\"oec-table__cell--strong\\">$661.28</td><td class=\\"oec-table__cell--strong oec-table__cell--red\\">($661.28)</td></tr></tbody></table>"`;
